src/main/java/de/zalando/aruha/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -50,6 +50,9 @@ public void onEnter() {
     }
 
     private void onTopologyChanged() {
+        if (topologyListener == null) {
```

Comment:
NullPointerException is not really informative, I think IllegalStateException is a perfect description of what happens here, the ClosingState object is in incorrect state where this method is called.